src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -2784,11 +2784,15 @@ StackFrame ExInfo::FindParentStackFrameHelper(CrawlFrame* pCF,
     // Check for out-of-line finally funclets.  Filter funclets can't be out-of-line.
     if (!fIsFilterFunclet)
     {
-        if (pRegDisplay->IsCallerContextValid)
-        {
-            PCODE callerIP = dac_cast<PCODE>(GetIP(pRegDisplay->pCallerContext));
-            BOOL fIsCallerInVM = FALSE;
+        PCODE callerIP = dac_cast<PCODE>(GetIP(pRegDisplay->pCallerContext));
+        BOOL fIsCallerInVM = FALSE;
 
+        if (callerIP == InterpreterFrame::DummyCallerIP)
```

Comment:
Nit: If you make the special dummy value to be an address in VM, we won't need this special check - it will fall out naturally from the logic below.